contracts/VestedEscrow.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+/*
+Rewrite of Curve Finance's Vested Escrow
+found at https://github.com/curvefi/curve-dao-contracts/blob/master/contracts/VestingEscrow.vy
+
+Changes:
+- no disable methods
+- only one fund admin
+- add claim and stake
+*/
+import "./helper/MathUtil.sol";
+import "./interfaces/IRewards.sol";
```

Comment:
Should be `import "./Interfaces/IRewards.sol";` ci fails because that

Reply:
But yes, after that fix locally deploy stops for me after (was checking on angle)


```
   Deploying 'TransparentUpgradeableProxy'
   ---------------------------------------
   > transaction hash:    0xc6589a9e40f159ccd990674ad5cb221b38d82d9c219cffaa1d5c314fae51bdf0
   > Blocks: 0            Seconds: 0
   > contract address:    0x7311Eeed39eEa62B1d4C102a4BEf7b0FEAb06B91
   > block number:        15181022
   > block timestamp:     1658340502
   > account:             0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1
   > balance:             999988.923095116636443
   > gas used:            824655 (0xc954f)
   > gas price:           50 gwei
   > value sent:          0 ETH
   > total cost:          0.04123275 ETH


Exiting: Review successful transactions manually by checking the transaction hashes above on Etherscan.


Error: Invalid JSON RPC response: ""
    at Object.InvalidResponse (/newve/vested/veToken-contracts/node_modules/web3-core-helpers/lib/errors.js:43:16)
    at XMLHttpRequest.request.onreadystatechange (/newve/vested/veToken-contracts/node_modules/web3-providers-http/lib/index.js:95:32)
    at XMLHttpRequestEventTarget.dispatchEvent (/newve/vested/veToken-contracts/node_modules/xhr2-cookies/xml-http-request-event-target.ts:44:13)
    at XMLHttpRequest._setReadyState (/newve/vested/veToken-contracts/node_modules/xhr2-cookies/xml-http-request.ts:219:8)
    at XMLHttpRequest._onHttpRequestError (/newve/vested/veToken-contracts/node_modules/xhr2-cookies/xml-http-request.ts:379:8)
    at ClientRequest.<anonymous> (/newve/vested/veToken-contracts/node_modules/xhr2-cookies/xml-http-request.ts:266:37)
    at ClientRequest.emit (node:events:527:28)
    at Socket.socketErrorListener (node:_http_client:454:9)
    at Socket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Truffle v5.5.18 (core: 5.5.18)
Node v16.16.0
```
Didn't had change to look more, yet

Reply:
Fixed the import, just running `npm run deploy-basic-contract` seems to work fine here using fork-mainnet so you might be having an issue with the angle fork or something

Reply:
Sorry, for delay.
@Jiverrr You still need fix deployment for angle and idle, as basically if make it to stay as it right now, it would break angle and idle deployment and those projects are first to be launched. Also, it would break full CI workflow and all other tests. There no point to test in case its not working with current angle and idle deployments

Overall, looks ok, and test passes with fork-mainnet and only deploy-basic-contract, but we couldn't accept it in current state, would require to modify migrations anyway
Please take a look at [dumpAddresses.js](https://github.com/veTokenFinance/veToken-contracts/blob/main/test/helper/dumpAddresses.js) file as well

@c-future Am I right? Please, comment on that matter

Reply:
@solvetony I have found the issue, the problem was setting the start time using 'time' from the OpenZeppelin test helpers which didn't seem to work here, weirdly enough it did work on fork mainnet for some reason. I have set the start time using a javascript function now instead and added a comment that it should be configured later when we have decided on the time to start vesting, or it can be updated later from within the contract. 

The tests are succeeding now

Reply:
@c-future Please, merge. Seems alright!